Update: swarm support for ansible-traefik-docker

till opened the pull request:
- introduces new vars to control if we run on swarm
    - `traefik_use_swarm`
    - (optional) `traefik_swarm_network_opts` 
    - `traefik_compose_version` (must be set to `3` for Swarm)

till committed:
Update: swarm support for ansible-traefik-docker
  - introduces new vars to control if we run on swarm
    - `use_swarm`
    - `traefik_compose_version`

rwos commented on Makefile:
```diff
@@ -11,4 +11,4 @@ test: build-testimage
 		-w /tmp/${ROLENAME} \
 		${TESTIMAGENAME} \
-		sudo molecule test
+		sudo molecule test --scenario-name swarm
```
just so we don't forget it - this needs to be changed to "run all scenarios" (not sure how yet :D)

rwos replied:
`molecule test --all` does that apparently

rwos commented on defaults/main.yml:
```diff
@@ -34,3 +36,7 @@ traefik_consul_enable: false
+
+# Compose, or Swarm?
+use_swarm: false
+swarm_network_opts: "--attachable --driver=overlay"
```
shouldn't we prefix those with "traefik_" or something? Not sure what the ansible best practices are in that regard

rwos commented on templates/docker-compose.yml.j2:
```diff
@@ -1,5 +1,5 @@
 # {{ ansible_managed }}
-version: '2'
+version: '{{ traefik_compose_version }}'
```
this works, but just always setting it to `'3'` is maybe easier?

till replied:
@rwos I am not sure, theoretically yes, but it's also just a weird hack.

till replied:
Best would be to override the template completely, but I am not sure I want to add that right now.

till committed:
Fix: run all scenarios

till committed:
Update: prefix variables

till committed:
Refactor: tests

 - add explicit version (to be tested later)
 - move default/playbook prep into prepare.yml
 - fix: assert "expected" "actual" syntax
 - rename test func to know exactly where it failed
 - fix: install explicit docker version (by moving vars into prepare)

till commented on molecule/swarm/tests/test_default.py:
```diff
@@ -0,0 +1,44 @@
+    out = host.check_output(
+        'docker ps --filter "label=traefik.backend=traefik"'
+        + ' --format {%raw%}"{{.Image}}"{%endraw%}')
+    assert 'traefik:latest' == out
```
This actually responds different on Swarm. Go figure. 🤦‍♂️

till committed:
Refactor: split test suite into two jobs

till committed:
Update: keep artifacts around

till committed:
Fix: compose file to publish all ports
